src/hooks/useWithdrawLiquidityHandler.tsx

```diff
@@ -1,372 +1,251 @@
+      }
+    } catch (err: any) {
+      notifyError(err.toString())
     }
   }
 
-  async function approveTransaction() {
-    // Position Manager contract address
-    let address =
-      process.env.NEXT_PUBLIC_ENVIRONMENT === 'local'
-        ? process.env.NEXT_PUBLIC_POSITION_MANAGER_ADDRESS || ''
-        : DEPOSIT_POOL_ADDRESS
+  useEffect(() => {
     if (provider) {
-      if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-        if (accountInfo?.address) {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider.getSigner(accountInfo.address)))
-        } else {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider))
-        }
-      } else {
-        // if (accountInfo?.address) {
-        //   setDepPool(new ethers.Contract(address, DepPool.abi, provider.getSigner(accountInfo.address)))
-        // } else {
-        //   setDepPool(new ethers.Contract(address, DepPool.abi, provider))
-        // }
+      setInvalidBtnText('Confirm')
+      if (!positionManager && POSITION_MANAGER_ADDRESS) {
+        setPositionManager(
+          new ethers.Contract(
+            POSITION_MANAGER_ADDRESS,
+            PositionManagerJSON.abi,
+            accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+          )
+        )
       }
-    } else {
-      notifyError('Please connect wallet')
-    }
 
-    if (!accountInfo || !accountInfo.address) {
-      notifyError('Wallet not connected.')
-      return
+      if (!gammaPoolContract && poolAddress) {
+        setGammaPoolContract(
+          new ethers.Contract(
+            poolAddress.toString(),
+            GammaPool.abi,
+            accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+          )
+        )
+      }
     }
-    if (accountInfo && accountInfo?.address) {
-      if (posManager === null) {
-        return
+  }, [provider])
+
+  async function approveTransaction() {
+    try {
+      let approval
+
+      if (!gammaPoolContract) {
+        await initializeGammaPoolContract()
       }
-      approveWithdraw(posManager, posManager.address)
-        .then(() => {
-          setEnableRemove(true)
-        })
-        .catch((err: any) => {
-          setEnableRemove(false)
-          notifyError(err)
-        })
-    } else {
-      if (depPool === null) {
-        return
+
+      if (gammaPoolContract && poolAddress && accountInfo?.address && provider) {
+        let amt = '0'
+        if (sliderPercentage === 0) {
+          throw new Error('Please select a valid amount to withdraw')
+        }
+        if (sliderPercentage === 100) {
+          amt = totalLiquidityAmt.toString()
+        } else {
+          amt = ((liquidityAmt * sliderPercentage) / 100).toString().toString()
+        }
+        approval = await validateAllowance(
+          accountInfo.address,
+          gammaPoolContract,
+          BigNumber.from(amt.toString()),
```

Comment:
I'm getting error on this line Argument of type 'BigNumber' is not assignable to parameter of type 'number'..

Reply:
nvm